Fixed #30573 -- Rephrased documentation to avoid words that minimise the involved difficulty.

rixx opened the pull request:
Obviously (heh.) this patch does not remove all occurrences of the words
in question. Rather, I went through all of the occurrences of the words
listed below, and judged if they a) suggested the reader had some kind
of knowledge/experience, and b) if they added anything of value
(including tone of voice, etc). I left most of the words alone. I looked
at the following words:

- simply/simple
- easy/easier/easiest
- obvious/obviously
- just
- merely
- straightforward
- ridiculous

Thanks to Carlton Gibson for guidance on how to approach this issue, and to Tim Bell for providing the idea.

Ticket link: https://code.djangoproject.com/ticket/30573

felixxm commented on docs/faq/contributing.txt:
```diff
@@ -84,16 +84,16 @@ of a larger problem. While we can spend time writing, testing and applying
-bug regularly, it doesn't necessarily follow that every single Django user
-will hit the same bug. Different users use Django in different ways, stressing
+bug regularly, it doesn't necessarily follow that every single Django user will
+hit the same bug. Different users use Django in different ways, stressing
```
This patch is quite big, can you revert all unrelated changes like this :pray:

rixx replied:
Hmm, I rewrapped all text I changed to 80 characters, because I thought docs were supposed to have consistent wrapping, but yeah, I can try to revert those.

felixxm replied:
Documentation should be wrapped at 79 chars (see [#python-style](https://docs.djangoproject.com/en/2.2/internals/contributing/writing-code/coding-style/#python-style)). But even if it's not wrapped in some places, please don’t waste time doing unrelated refactoring of existing documentation to conform to this style.

rixx replied:
This will leave a couple of weird short lines in the middle of text, but I guess it's preferable for reviewing purpose. Sorry for the noise.

felixxm replied:
So to be clear if you for example remove word `easy` from some paragraph then yes we should fix wrapping for the entire paragraph at 79 chars. I'm only talking about omitting totally unrelated changes like with this `will`.

rixx replied:
I think I made some changes to that paragraph and reverted them, and by accident the wrapping stayed in. I tried to touch only the paragraphs I changed. Going to revert this change.

rixx replied:
I went through the patch and made sure no further unrelated changes were made, and changed the wrapping to change paragraphs I touched only from the point of change, not above that.

carltongibson commented:
Hey @rixx. Thanks for this. (Can't remember what I actually said, but 👍 🙂)

> Obviously...

Perfect. 😀

@claudep, @felixxm makes a good point on the ticket about creating work for the translation teams (so not to backport). Tim often would mention the need for re-translations too. Can I ask, what's the general maxim there? And, for me, this patch seems positive and in line with what we aim for generally, so is it reasonable, without putting too much burden on our current volunteers? (I'm just not sure of the state of play. Thank you!)

rixx commented:
> Tim often would mention the need for re-translations too. Can I ask, what's the general maxim there? And, for me, this patch seems positive and in line with what we aim for generally, so is it reasonable, without putting too much burden on our current volunteers?

I have to admit that I hadn't thought of the additional burdon for translators. If you judge this PR to have an unhelpful ratio of effort for translators as opposed to its usefulness, we can definitely decide to close it.

carltongibson commented:
I think the sentiment, that we eliminate the "just" and "obviously" and so on is right. And it can't be that we can't make _improvements_ — but knowing the work it involves is helpful to judge. Obviously (🙂) there's no hard line; I was fishing for rule-of-thumb guidance... (Thanks again.)

claudep commented:
If this is really improving the docs (not simple nit-picking), you should do it. I just say that the stable docs should only be changed for real errors, because between the push and the translations going on-line, there is a gap of ~1 month where those strings appear untranslated.

carltongibson commented:
OK, thanks @claudep. That's a good guiding principle.

adamchainz commented on docs/howto/deployment/wsgi/gunicorn.txt:
```diff
@@ -21,16 +21,15 @@ Running Django in Gunicorn as a generic WSGI application
-`application`. So for a typical Django project, invoking gunicorn would look
-like::
+the Gunicorn server process. At its simplest, gunicorn is called with the
+location of a module containing a WSGI application object named `application`.
```
"is called" is passive voice and doesn't read well here. Perhaps:

"The simplest invocation of gunicorn is to pass only the name of a module containing a WSGI application called `application`, which for a typical Django project would look like:"

rixx replied:
Much better, thank you.

adamchainz commented on docs/howto/deployment/wsgi/gunicorn.txt:
```diff
@@ -21,16 +21,15 @@ Running Django in Gunicorn as a generic WSGI application
-It requires that your project be on the Python path; the simplest way to ensure
-that is to run this command from the same directory as your ``manage.py`` file.
+It requires that your project be on the Python path; you can ensure this by
+running this command from the same directory as your ``manage.py`` file.
```
I think something was lost here. This isn't the *only* way of ensuring the project is on the python path, it's just the simplest way. I prefer the old wording. Again "simple" here doesn't seem to be a comment on the reader's experience level

adamchainz commented on docs/howto/static-files/deployment.txt:
```diff
@@ -93,8 +94,8 @@ For example, if you've written an S3 storage backend in
 static files would be pushed through your storage package up to S3. If you
-later needed to switch to a different storage provider, it could be as simple
-as changing your :setting:`STATICFILES_STORAGE` setting.
+later needed to switch to a different storage provider, you may just have to
```
I think "only" is a better word here than "just"

adamchainz commented on docs/howto/windows.txt:
```diff
@@ -57,7 +57,7 @@ Install ``virtualenv`` and ``virtualenvwrapper``
 each Django project you create. While not mandatory, this is considered a best
 practice and will save you time in the future when you're ready to deploy your
-project. Simply type::
+project::
```
I think we do need a leading sentence like. "To do this, run:"

adamchainz commented on docs/internals/contributing/bugs-and-features.txt:
```diff
@@ -46,13 +46,13 @@ particular:
-  |django-updates|, which is tracked by developers and interested
-  community members; we see them as they are filed.
+* **Don't** post to |django-developers| to announce that you have filed a bug
+  report. All the tickets are mailed to another list, |django-updates|, which
```
I think something was lost in removing "just". Perhaps substitute with "only". For tickets that need some community debate, it is legitimate to post to django-developers after filing, but the post should be more than "I filed a ticket".

adamchainz commented on docs/internals/contributing/committing-code.txt:
```diff
@@ -11,8 +11,8 @@ contribute code to Django, look at :doc:`writing-code/working-with-git` instead.
 
-Since Django is now hosted at GitHub, most patches are provided in the form of
-pull requests.
+Since Django is hosted at GitHub, patches are provided in the form of pull
```
I'd say it's hosted "on" GitHub... ?

rixx replied:
Yeah, I stumbled over that, too, but I didn't want to make unrelated changes.

adamchainz replied:
As above I think it's fine to avoid retranslation

adamchainz commented on docs/internals/contributing/committing-code.txt:
```diff
@@ -26,14 +26,14 @@ builders that doesn't run automatically, such as Oracle or Selenium. See the
-Now you can simply run ``git pr ####`` to checkout the corresponding pull
-request.
+Add it to your ``~/.gitconfig``, and set ``upstream`` to be ``django/django``.
+Now you can run ``git pr ####`` to checkout the corresponding pull request.
```
"Now" -> "Then" - it's not *now* until the user has actually done it :)

rixx commented on docs/faq/general.txt:
```diff
@@ -61,7 +61,7 @@ Django uses a "shared-nothing" architecture, which means you can add hardware
 
 The framework cleanly separates components such as its database layer and
-application layer. And it ships with a simple-yet-powerful
+application layer. And it ships with a light-yet-powerful
```
Good point, I reverted that change!

I tried to keep "simple" whereever it didn't concern the PR topic, but after some point I probably developed blind spots.

rixx commented on docs/faq/usage.txt:
```diff
@@ -61,9 +61,9 @@ Using a :class:`~django.db.models.FileField` or an
-be dynamically-generated menus. At first glance, it seems logical to simply
-add a common dictionary to the template context.
+Sometimes your templates all need the same thing. A common example would be
+dynamically-generated menus. At first glance, it seems logical to add a common
```
Is it alright to make related changes like that in the same commit?

adamchainz replied:
I think it is for docs since it prevents re-translation.

rixx commented on docs/howto/custom-model-fields.txt:
```diff
@@ -81,11 +81,12 @@ Background theory
+something more complex like ``Hand`` -- and convert it to and from a format
+that is useful when dealing with the database (and serialization, but, as we'll
+see later, that falls out fairly naturally once you have the database side
+under control).
```
Definitely a helpful change, I applied it. Is "that falls out fairly naturally …" idiomatic English? I don't think I'm familiar with the usage of "falling out" in that context.

adamchainz replied:
Yeah it's not so common. Maybe "that is easier once we have the database side under control." ?

adamchainz commented on docs/ref/models/instances.txt:
```diff
@@ -749,9 +749,9 @@ This template code is much better:
-for something simple such as correcting a spelling error, you don't want to
-have to track down every place that the URL might be created. Specify it once,
-in ``get_absolute_url()`` and have all your other code call that one place.
+for something like correcting a spelling error, you don't want to have to track
```
I think "for something like correcting a spelling error" is *implying* that this is a simple task without saying it, and it's better to have the word in there then. Perhaps "small" rather than "simple" ?

adamchainz commented on docs/ref/request-response.txt:
```diff
@@ -167,7 +167,7 @@ All attributes should be considered read-only, unless stated otherwise.
     Web servers like Nginx and Apache 2.4+.
 
-    :attr:`HttpRequest.headers` is a simpler way to access all HTTP-prefixed
+    :attr:`HttpRequest.headers` is another way to access all HTTP-prefixed
```
I think this can be reverted, it's simpler and it's intended to be used because it is simpler

adamchainz commented on docs/topics/auth/customizing.txt:
```diff
@@ -82,8 +82,8 @@ backends that follow.
-    you need to force users to re-authenticate using different methods. A simple
-    way to do that is simply to execute ``Session.objects.all().delete()``.
+    you need to force users to re-authenticate using different methods. A way
+    to do that is to execute ``Session.objects.all().delete()``.
```
I think the sentence can start with "A simple way" still - there are certainly more complicated methods. The "simply" was definitely redundant though.

adamchainz commented on docs/topics/auth/default.txt:
```diff
@@ -150,7 +150,7 @@ Authenticating users
 =============================
 
-Django comes with a simple permissions system. It provides a way to assign
+Django comes with an inbuilt permissions system. It provides a way to assign
```
"builtin" is the phrase used in the docs, "inbuilt" isn't used anywhere at the moment.

adamchainz commented on docs/topics/class-based-views/mixins.txt:
```diff
@@ -481,7 +479,7 @@ Our new ``AuthorDetail`` looks like this::
             return super().form_valid(form)
 
-``get_success_url()`` is just providing somewhere to redirect to,
+``get_success_url()`` is providing some place to redirect to,
```
"provides somewhere to redirect to"

adamchainz commented on docs/topics/db/managers.txt:
```diff
@@ -145,8 +145,8 @@ So these statements are all legal::
 managers on the same model. You can attach as many ``Manager()`` instances to
-a model as you'd like. This is an easy way to define common "filters" for your
-models.
+a model as you'd like. This is an non-repetitive way to define common "filters"
```
should be "a" not "an" before "non-repetitive"

adamchainz commented on docs/topics/db/managers.txt:
```diff
@@ -446,10 +446,9 @@ following code must work::
-method of your ``Manager`` object that controls object state, you
-should ensure that you don't affect the ability of your ``Manager`` to
-be copied.
+This won't be an issue for most custom managers. If you are adding methods to
```
I think this could be reverted or "just" replaced with "only"

adamchainz commented on docs/topics/db/optimization.txt:
```diff
@@ -217,7 +217,7 @@ Don't retrieve things you don't need
 -----------------------------------------------
 
-When you just want a ``dict`` or ``list`` of values, and don't need ORM model
+When you want a ``dict`` or ``list`` of values, and don't need ORM model
```
"only" would be good

adamchainz commented on docs/topics/db/optimization.txt:
```diff
@@ -260,7 +260,7 @@ But:
 ------------------------------------------
 
-If you are going to need other data from the QuerySet, just evaluate it.
+If you are going to need other data from the QuerySet, evaluate it.
```
"evaluate it immediately." ?

adamchainz commented on docs/topics/db/transactions.txt:
```diff
@@ -622,9 +622,9 @@ Handling exceptions within PostgreSQL transactions
-until end of transaction block". While simple use of ``save()`` is unlikely
-to raise an exception in PostgreSQL, there are more advanced usage patterns
-which might, such as saving objects with unique fields, saving using the
+until end of transaction block". While the use of ``save()`` is unlikely to
```
Maybe "basic use of save()"? `save()` is still used in the following "more advanced usage patterns"

adamchainz commented on docs/topics/email.txt:
```diff
@@ -45,8 +44,7 @@ a secure connection is used.
 
-The simplest way to send email is using
-``django.core.mail.send_mail()``.
+In most cases, you can send mails using ``django.core.mail.send_mail()``.
```
"send email" I think

adamchainz commented on docs/topics/forms/formsets.txt:
```diff
@@ -651,9 +651,9 @@ This is useful if you want to :ref:`use more than one formset in a view
-Using a formset inside a view is as easy as using a regular ``Form`` class.
-The only thing you will want to be aware of is making sure to use the
-management form inside the template. Let's look at a sample view::
+Using a formset inside a view is not much different from using a regular
```
"not very different" reads better

adamchainz commented on docs/topics/forms/index.txt:
```diff
@@ -181,8 +181,8 @@ Building a form
-Suppose you want to create a simple form on your website, in order to obtain
-the user's name. You'd need something like this in your template:
+Suppose you want to create a form on your website, in order to obtain the
+user's name. You'd need something like this in your template:
```
I think this one can be reverted, it reads weird without the "simple"

adamchainz commented on docs/topics/forms/index.txt:
```diff
@@ -206,10 +206,10 @@ will contain the form data.
-hundreds of fields, many of which might need to be pre-populated, and we might
-expect the user to work through the edit-submit cycle several times before
-concluding the operation.
+In practice, a form might contain dozens or hundreds of fields instead of just
```
I think this can be reverted, the missing sentence makes the rest read weirdly because it's only the second half of an implicit comparison.

adamchainz commented on docs/topics/forms/media.txt:
```diff
@@ -2,7 +2,7 @@
 =================================
 
-Rendering an attractive and easy-to-use Web form requires more than just
+Rendering an attractive and easy-to-use Web form requires more than
```
I think this one can be reverted, it has the sense of "only" and "only" doesn't sound as good

adamchainz commented on docs/topics/forms/modelforms.txt:
```diff
@@ -753,8 +752,8 @@ Model formsets
 Like :doc:`regular formsets </topics/forms/formsets>`, Django provides a couple
-of enhanced formset classes that make it easy to work with Django models. Let's
-reuse the ``Author`` model from above::
+of enhanced formset classes that work well with Django models. Let's reuse the
```
"work well" is a bit of a value judgment. I think "to make working with Django models more convenient." carries the intended meaning better.

adamchainz commented on docs/topics/http/sessions.txt:
```diff
@@ -701,9 +700,9 @@ In order to build a custom session engine or to customize an existing one, you
-the next section for details).
+You can generally extend the session engines, but doing so with database-backed
+session engines generally requires some extra effort (see the next section for
+details).
```
Double "generally" doesn't read well. I think dropping the first one still reads fine: "You can extend the session engines, but doing so..."

adamchainz commented on docs/topics/http/urls.txt:
```diff
@@ -567,7 +567,7 @@ directions:
 The first one is the usage we've been discussing in the previous sections. The
 second one is what is known as *reverse resolution of URLs*, *reverse URL
-matching*, *reverse URL lookup*, or simply *URL reversing*.
+matching*, *reverse URL lookup*, or *URL reversing*.
```
I think this can be reverted, it's highlighting the term that will be used in the rest of the page/documentation

adamchainz commented on docs/topics/http/views.txt:
```diff
@@ -60,12 +60,12 @@ date and time. To display this view at a particular URL, you'll need to create a
 ================
 
-Returning HTTP error codes in Django is easy. There are subclasses of
+Django provides help for returning HTTP error codes: There are subclasses of
```
Full stop (period) instead of colon would make the sentences shorter, easier to read, and easier to translate.

adamchainz commented on docs/topics/i18n/translation.txt:
```diff
@@ -1195,8 +1195,8 @@ a word or not::
     // true
 
-In the simplest case, if no custom pluralization is needed, this returns
-``false`` for the integer ``1`` and ``true`` for all other numbers.
```
This *is* the simplest case, so I think revert it. Again not a judgment of the reader.

adamchainz commented on docs/topics/install.txt:
```diff
@@ -221,8 +219,8 @@ latest bug fixes and improvements, follow these instructions:
-Git will automatically download any changes.
+When you want to update your copy of the Django source code, run the command
+``git pull`` from within the ``django`` directory. When you do this, Git will
+automatically download any changes.
```
I'd drop the "automatically" here - it's not automatic, we just manually triggered it!

adamchainz commented on docs/topics/logging.txt:
```diff
@@ -198,7 +198,7 @@ There are two other logging calls available:
 ===================
 
-Of course, it isn't enough to just put logging calls into your code.
+Of course, it isn't enough to put logging calls into your code.
```
I think revert, this has the sense "only"

adamchainz commented on docs/topics/migrations.txt:
```diff
@@ -114,8 +114,8 @@ Django projects without the need for a full database.
 
-Working with migrations is simple. Make changes to your models - say, add
-a field and remove a model - and then run :djadmin:`makemigrations`::
+Django will create migrations for you. Make changes to your models - say, add a
```
I'd say "Django can create migrations for you." - it won't if you don't tell it, and you have the option to create them yourself too.

adamchainz commented on docs/topics/migrations.txt:
```diff
@@ -660,14 +657,14 @@ You must then transition the squashed migration to a normal migration by:
-can't serialize out into a valid Python representation - there's no Python
-standard for how a value can be turned back into code (``repr()`` only works
-for basic values, and doesn't specify import paths).
+While Django can serialize most things, there are some things that we can't
```
I'd revert this one, it's used for emphasis

adamchainz commented on docs/topics/pagination.txt:
```diff
@@ -238,7 +238,7 @@ you can catch either of the following exceptions:
 
 Both of the exceptions are subclasses of :exc:`InvalidPage`, so you can handle
-them both with a simple ``except InvalidPage``.
+them both with an ``except InvalidPage``.
```
I'd even drop the "an"

adamchainz commented on docs/topics/performance.txt:
```diff
@@ -49,8 +49,7 @@ and for that you need:
 
-It's no good just guessing or assuming where the inefficiencies lie in your
-code.
+It's no good guessing or assuming where the inefficiencies lie in your code.
```
I'd revert this one, again it's for emphasis

adamchainz commented on docs/topics/serialization.txt:
```diff
@@ -15,7 +15,7 @@ serializer to handle any format (text-based or not).
 ================
 
-At the highest level, serializing data is a very simple operation::
+At the highest level, are used like this::
```
Missing subject. Perhaps: "At the highest level, serializing data can be done like this::"

rixx replied:
"At the highest level, you can serialize data like this::" to avoid passive voice?

adamchainz replied:
Yes good point!

adamchainz commented on docs/topics/serialization.txt:
```diff
@@ -241,13 +241,13 @@ JSON in the following way::
-The formatting here is a bit simpler than with XML. The whole collection
-is just represented as an array and the objects are represented by JSON objects
-with three properties: "pk", "model" and "fields". "fields" is again an object
+The formatting here is less complex than with XML. The whole collection is
```
i'd revert this one, "a bit simpler" and "less complex" are two ways of saying the same thing

rixx commented on docs/intro/contributing.txt:
```diff
@@ -10,7 +10,7 @@ in Django that you'd like to see fixed, or maybe there's a small feature you
 
 Contributing back to Django itself is the best way to see your own concerns
-addressed. This may seem daunting at first, but it's really pretty simple.
+addressed. This may seem daunting at first, but it's easier than it seems.
```
This sounds lovely, yes :)

rixx commented on docs/ref/contrib/admin/index.txt:
```diff
@@ -2996,9 +2994,9 @@ returns a site instance.
-Django-powered website. Just create multiple instances of ``AdminSite`` and
-root each one at a different URL.
+You can create multiple instances of the admin site on the same Django-powered
+website. Create multiple instances of ``AdminSite`` and root each one at a
```
Installing pages at URLs sounds unfamiliar to me. Maybe "place each one at a different URL"?

adamchainz replied:
"Placing" sounds just fine.

rixx commented on docs/ref/contrib/gis/functions.txt:
```diff
@@ -234,7 +234,7 @@ the geometries (e.g. `ST_DistanceSphere
 When distances are calculated with geodetic (angular) coordinates, as is the
 case with the default WGS84 (4326) SRID, you can set the ``spheroid`` keyword
-argument to decide if the calculation should be based on a simple sphere (less
+argument to decide if the calculation should be based on a sphere (less
```
You're right, of course. I generally tried to touch the GIS docs as little as possible, because "simple" often referred to geometric properties.

rixx commented:
I *think* I addressed all your review comments by resolving them, except for two which I replied to. Thank you so much for reviewing this little monster!

adamchainz commented:
Nice work!

rixx commented:
Right, got the rebase conflict addressed, too, so this is looking good, pending further review :)

rixx commented:
I'm completely happy with your review, I just saw that there was "1 pending reviewer" in the GitHub interface, so I wasn't sure how this was going to proceed. I don't need a second review ;)

carltongibson commented:
>  ...just saw that there was "1 pending reviewer"... 

I want to go through this, so it's that's me. It's on the list for this week. 🙂

rixx commented:
> Hi @rixx.
> 
> Just before diving into the individual changes, I think, having gone to this effort, we should add a paragraph/sentence to `writing-documentation.txt` in the _Writing style_ section, saying to avoid using _simple_ etc in this way. What do you think?

Absolutely, I added a change like that. Feel free to correct the phrasing, as always.

carltongibson commented:
Right, roll-up sleeves... 

I rebased and pulled the Style suggestion into a separate commit cf9e66fe58e198faf647a07b2ab5625dba417985. (@felixxm We could pull that in already no?) 

I'll go through the text changes one-by-one now.

felixxm commented:
@carltongibson Sure, merged and backported (I fixed one typo).

carltongibson commented:
Thanks @felixxm!

rixx commented on docs/ref/csrf.txt:
```diff
@@ -248,8 +248,8 @@ The CSRF protection is based on the following things:
 
-   In order to protect against `BREACH`_ attacks, the token is not the secret;
-   a random salt is prepended to the secret and used to scramble it.
+   In order to protect against `BREACH`_ attacks, the token is not simple the
```
simply, not simple.

carltongibson replied:
Good catch. Thanks.

rixx commented:
@carltongibson Thank you very much for your review! Apart from one typo, your changes look fine to me.

I also sympathised with "just" and "simple", and left many of them in place, but I appreciate your reverts. I guess after some time staring at the same words, I leaned too much on the side of removal. Regarding the cosmetic rewraps: I tried to remove all of them and just rewrap paragraphs ends following my changes. If you'd like me to, I can check the PR again and see what should be reverted?

carltongibson commented:
Typo fixed. 👍 

> Regarding the cosmetic rewraps: I tried to remove all of them and just rewrap paragraphs ends following my changes. If you'd like me to, I can check the PR again and see what should be reverted?

If you have energy... They weren't _incorrect_ so I left them. At this point I'd be happy to see them left. (@felixxm may have an opinion, which he'll state if he does 🙂)

> ...I guess after some time staring at the same words...

Yes. 😀

Much effort. Hard to do. Hard to review. 

You've done a great job here.

felixxm commented:
buildbot

felixxm commented:
Thanks y'all :star2:
